src/components/common/DatePicker/DatePicker.component.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import ArrowLeft from '@/assets/svg/chevron-left.svg';
+import ArrowRight from '@/assets/svg/chevron-right.svg';
+
+import * as Styled from './DatePicker.styled';
+
+const handleGenerateDateRows = (date: Dayjs) => {
+  const { accRows: rows } = Array.from({ length: date.daysInMonth() }).reduce<{
+    accRows: Dayjs[][];
+    accRowIndex: number;
+  }>(
+    ({ accRows, accRowIndex }, _, index) => {
+      const currentIndexDate = dayjs(new Date(date.year(), date.month(), index + 1));
+
+      accRows[accRowIndex].push(currentIndexDate);
+
+      if (currentIndexDate.day() === 6 && index !== date.daysInMonth() - 1) {
+        return { accRows: [...accRows, []], accRowIndex: accRowIndex + 1 };
+      }
+
+      return { accRows, accRowIndex };
+    },
+    { accRows: [[]], accRowIndex: 0 },
+  );
+
+  return rows.map((each, index) => {
+    if (index === 0 && each.length !== 7) {
+      return [
+        ...Array.from({ length: 7 - each.length }).map((_, i) =>
+          each[0].clone().startOf('week').add(i, 'day'),
+        ),
+        ...each,
+      ];
+    }
+
+    if (index === rows.length - 1 && each.length !== 7) {
+      return [
+        ...each,
+        ...Array.from({ length: 7 - each.length }).map((_, i) =>
+          each[each.length - 1].clone().add(i + 1, 'day'),
+        ),
+      ];
+    }
+
+    return each;
+  });
+};
+
+interface DayCellProps extends React.TdHTMLAttributes<HTMLElement> {
+  date: Dayjs;
+  current: Dayjs;
+  selectedDate: Dayjs | null;
+  disablePast: boolean;
+}
+
+const CALENDAR_COLUMN = ['일', '월', '화', '수', '목', '금', '토'];
+
+const DayCell = ({
+  date,
+  current,
+  selectedDate,
+  disablePast = false,
+  ...resetProps
+}: DayCellProps) => {
+  const today = dayjs();
+  const isCurrentMonth = date.month() === current.month();
+  const isDisabled =
+    (disablePast && date.isBefore(today) && !date.isSame(today, 'day')) || !isCurrentMonth;
+
+  const props = {
+    ...resetProps,
+    disabled: isDisabled,
+    today: date.isSame(today, 'day'),
+    selected: date.isSame(selectedDate),
+  };
+
+  return <Styled.DatePickerTd {...props}>{date.format('D')}</Styled.DatePickerTd>;
+};
+
+export interface DatePickerProps {
+  handleSelectDate: (clickedDate: Dayjs) => void;
+  selectedDate: Dayjs | null;
+}
+
+const DatePicker = ({ handleSelectDate, selectedDate }: DatePickerProps) => {
+  const [date, setDate] = useState<Dayjs>(dayjs());
```

Comment:
date 기준 같은 월의 날짜들이 생성되어 캘린더에 그려지는 걸로 이해했는데 맞을까요?

맞다면 selectedDate가 있는 경우 selectedDate가 있는 월이 바로 보이면 좋을거 같은데,
이 부분 괜찮다면 selectedDate 로 date 생성하는건 어떨까요?

Reply:
오 맞습니다 반영할게요!

```suggestion
const DatePicker = ({ handleSelectDate, selectedDate }: DatePickerProps) => {
  const [date, setDate] = useState<Dayjs>(selectedDate || dayjs());
```

요렇게 적용될 것 같습니다